File: plop-templates/adapter/payout/src/servlet/AdapterServlet.java
package com.rbkmoney.{{adapterPayoutPackageCase bank_name}}.servlet;

import com.rbkmoney.damsel.withdrawals.provider_adapter.AdapterSrv;
import com.rbkmoney.woody.thrift.impl.http.THServiceBuilder;
import org.springframework.beans.factory.annotation.Autowired;

import javax.servlet.GenericServlet;
import javax.servlet.Servlet;
import javax.servlet.ServletConfig;
import javax.servlet.ServletException;
import javax.servlet.ServletRequest;
import javax.servlet.ServletResponse;
import javax.servlet.annotation.WebServlet;

import java.io.IOException;

/**
 * Endpoint.
 */
@WebServlet("/adapter/{{kebabCase bank_name}}/payout")
public class AdapterServlet extends GenericServlet {

    @Autowired
    private AdapterSrv.Iface payoutAdapterServiceLogDecorator;

    private Servlet servlet;

    @Override
    public void init(ServletConfig config) throws ServletException {
        super.init(config);
        servlet = new THServiceBuilder().build(AdapterSrv.Iface.class, payoutAdapterServiceLogDecorator);
    }

    @Override
    public void service(ServletRequest request, ServletResponse response) throws ServletException, IOException {
        servlet.service(request, response);
    }
}

File: plop-templates/adapter/payout/src/servlet/AdapterServlet.kt
package com.rbkmoney.{{adapterPayoutPackageCase bank_name}}.servlet

import com.rbkmoney.damsel.withdrawals.provider_adapter.AdapterSrv
import com.rbkmoney.woody.thrift.impl.http.THServiceBuilder
import javax.servlet.GenericServlet
import javax.servlet.Servlet
import javax.servlet.ServletConfig
import javax.servlet.ServletRequest
import javax.servlet.ServletResponse
import javax.servlet.annotation.WebServlet

{{#if_eq doc true}}
/**
 * Endpoint.
 */
{{/if_eq}}
@WebServlet("/adapter/{{kebabCase bank_name}}/payout")
class AdapterServlet(
    private val payoutAdapterServiceLogDecorator: AdapterSrv.Iface
) : GenericServlet() {

    private lateinit var servlet: Servlet

    override fun init(config: ServletConfig) {
        super.init(config)
        servlet = THServiceBuilder().build(AdapterSrv.Iface::class.java, payoutAdapterServiceLogDecorator)
    }

    override fun service(request: ServletRequest, response: ServletResponse) =
        servlet.service(request, response)
}
